tests/optimize/test_objective.py

```diff
@@ -0,0 +1,43 @@
+"""Unit tests for objective module."""
+import unittest
+
+import relentless
+
+class QuadraticObjective(relentless.optimize.ObjectiveFunction):
+    """Mock objective function used to test relentless.optimize.ObjectiveFunction"""
+
+    def __init__(self, x):
+        self.x = x
+
+    def compute(self):
+        val = (self.x.value-1)**2
+        grad = {self.x:2*(self.x.value-1)}
+        return self.Result(val, grad)
+
+    def design_variables(self):
+        return (self.x,)
+
+class test_ObjectiveFunction(unittest.TestCase):
+    """Unit tests for relentless.optimize.ObjectiveFunction"""
+
+    def test_compute(self):
+        """Test compute method"""
+        x = relentless.variable.DesignVariable(value=4.0)
+        q = QuadraticObjective(x=x)
+
+        res = q.compute()
+        self.assertAlmostEqual(res.value, 9.0)
+        self.assertAlmostEqual(res.gradient(x), 6.0)
+        self.assertEqual(res.gradient('x'), 0.0)
```

Comment:
This test confused me at first. Maybe make it another variable (rather than a string with the same name) to make the intent clear?